inputgraph/inputgraph.go

```diff
@@ -195,7 +195,14 @@ func (l *loader) handleFor(ctx context.Context, forStmt spec.ForStatement) error
 }
 
 func (l *loader) handleWait(ctx context.Context, waitStmt spec.WaitStatement) error {
-	return errors.Wrap(ErrUnableToDetermineHash, "wait not supported")
+	l.hasher.HashString("wait")
```

Comment:
This is in-lieu of the `HashCommand` call, right? I wonder if we should separate this in a similar function and co-locate the functions? I'm thinking that if we make changes to one, we might want to change the other too, and it's easier to spot that if the functions are next to each other.

Reply:
this might be a good idea, because we already have `HashCommand`, `HashVersion`; however those commands just take a `spec.Command`, or `spec.Version` which are both leaf nodes of the ast.

WAIT on the other hand has references to other `spec.Command`, which means we have to walk the tree, and would need to call back into inputgraph package.

I honestly think it's fine keeping this as-is.


However, I think the current code might have an issue, as it's not clear how the following different blocks:

```
WAIT
  RUN echo foo
END
```

and

```
WAIT
END
RUN echo foo
```

will produce a different hash.

I think we would an additional:

`HashString("end")` or `HashInt(len(waitStmt.Body))`

Reply:
Updated with a new function called `HashStatement` in favor of specific functions for `HashWait`, `HashIf`, etc.